test/source-updater.test.js

```diff
@@ -1,232 +1,769 @@
+    audio: 'mp4a.40.2',
+    video: 'avc1.4D001E'
+  });
+
+  assert.notOk(this.sourceUpdater.updating(), 'not updating by default');
+
+  this.sourceUpdater.appendBuffer('video', mp4Video(), () => {
+    assert.notOk(this.sourceUpdater.updating(), 'not updating after append');
+    this.sourceUpdater.appendBuffer('audio', mp4Audio(), () => {
+      assert.notOk(this.sourceUpdater.updating(), 'not updating after append');
+      done();
+    });
+    assert.ok(this.sourceUpdater.updating(), 'updating during append');
+  });
 
-  this.mediaSource.trigger('sourceopen');
-  sourceBuffer = this.mediaSource.sourceBuffers[0];
-  updater.appendBuffer(new Uint8Array([0]));
-  assert.equal(sourceBuffer.updates_.length, 1, 'ran an update');
-  assert.deepEqual(sourceBuffer.updates_[0].append, new Uint8Array([0]),
-                  'appended the bytes');
+  assert.ok(this.sourceUpdater.updating(), 'updating during append');
 });
 
-QUnit.test('supports abort', function(assert) {
-  let updater = new SourceUpdater(this.mediaSource, 'video/mp2t');
-  let sourceBuffer;
+QUnit.test('dispose aborts and clears out audio and video buffers', function(assert) {
+  this.sourceUpdater.createSourceBuffers({
+    audio: 'mp4a.40.2',
+    video: 'avc1.4D001E'
+  });
+
+  // while this maintains internal logic of source updater (knowing the properties), it is
+  // good for this test to verify that those properties are cleared out
+  assert.ok(this.sourceUpdater.audioBuffer, 'have an audio buffer');
+  assert.ok(this.sourceUpdater.videoBuffer, 'have a video buffer');
+
+  // Let the original aborts run so that we don't mock out any behaviors.
+  const origAudioAbort =
+    this.sourceUpdater.audioBuffer.abort.bind(this.sourceUpdater.audioBuffer);
+  const origVideoAbort =
+    this.sourceUpdater.videoBuffer.abort.bind(this.sourceUpdater.videoBuffer);
+  let abortedAudio = false;
+  let abortedVideo = false;
+
+  this.sourceUpdater.audioBuffer.abort = () => {
+    abortedAudio = true;
+    origAudioAbort();
+  };
+  this.sourceUpdater.videoBuffer.abort = () => {
+    abortedVideo = true;
+    origVideoAbort();
+  };
+
+  this.sourceUpdater.dispose();
+
+  assert.ok(abortedAudio, 'aborted audio');
+  assert.ok(abortedVideo, 'aborted video');
+  assert.notOk(this.sourceUpdater.audioBuffer, 'removed audioBuffer reference');
+  assert.notOk(this.sourceUpdater.videoBuffer, 'removed videoBuffer reference');
+});
+
+QUnit.test('no error passed by default in done callback', function(assert) {
+  const done = assert.async();
+
+  this.sourceUpdater.createSourceBuffers({
+    audio: 'mp4a.40.2'
+  });
+
+  this.sourceUpdater.appendBuffer('audio', mp4Audio(), (error) => {
+    assert.notOk(error, 'no error');
+    done();
+  });
+});
+
+// Firefox allows appending video to an audio buffer, for some reason.
+// TODO find an alternative for triggering an error
+QUnit.skip('audio source buffer error passed in done callback', function(assert) {
```

Comment:
maybe append some corrupted data instead of actual video/audio data?

Reply:
Good call, that worked.